locator/tablets.hh

```diff
@@ -302,6 +336,15 @@ struct load_stats {
     }
 };
 
+struct repair_scheduler_config {
+    // When set to false, auto repair is disabled
+    bool auto_repair_enabled = false;
+    // If the time since last repair is bigger than auto_repair_threshold
+    // seconds, the tablet is eligible for auto repair.
+    std::chrono::seconds auto_repair_threshold{10 * 24 * 3600};
+    bool operator==(const repair_scheduler_config&) const = default;
```

Comment:
why is `operator==` needed to be defined?

Reply:
We have: 

```
 bool operator==(const tablet_map&) const = default; 
```
for tablet_map

without it it fails to compile

```
./types/types.hh:499:13: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const abstract_type' for 1st argument
  499 | inline bool operator==(const abstract_type& x, const abstract_type& y)
      |             ^          ~~~~~~~~~~~~~~~~~~~~~~
./types/types.hh:568:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const data_value' for 1st argument
  568 | bool operator==(const data_value& x, const data_value& y);
      |      ^          ~~~~~~~~~~~~~~~~~~~
./schema/schema.hh:377:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const column_definition' for 1st argument
  377 | bool operator==(const column_definition&, const column_definition&);
      |      ^          ~~~~~~~~~~~~~~~~~~~~~~~~
./schema/schema.hh:404:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const column_mapping_entry' for 1st argument
  404 | bool operator==(const column_mapping_entry& lhs, const column_mapping_entry& rhs);
      |      ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./schema/schema.hh:430:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const column_mapping' for 1st argument
  430 | bool operator==(const column_mapping& lhs, const column_mapping& rhs);
      |      ^          ~~~~~~~~~~~~~~~~~~~~~~~~~
./schema/schema.hh:464:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const raw_view_info' for 1st argument
  464 | bool operator==(const raw_view_info&, const raw_view_info&);
      |      ^          ~~~~~~~~~~~~~~~~~~~~
./schema/schema.hh:985:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const schema' for 1st argument
  985 | bool operator==(const schema&, const schema&);
      |      ^          ~~~~~~~~~~~~~
./reader_permit.hh:66:13: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const reader_resources' for 1st argument
   66 | inline bool operator==(const reader_resources& a, const reader_resources& b) {
      |             ^          ~~~~~~~~~~~~~~~~~~~~~~~~~
/home/asias/src/cloudius-systems/scylla/seastar/include/seastar/core/reactor.hh:107:6: note: candidate function not viable: no known conversion from 'const repair_scheduler_config' to 'const ::sockaddr_in' for 1st argument
  107 | bool operator==(const ::sockaddr_in a, const ::sockaddr_in b);
      |      ^          ~~~~~~~~~~~~~~~~~~~~~
/home/asias/src/cloudius-systems/scylla/seastar/include/seastar/core/shared_ptr.hh:733:1: note: candidate template ignored: could not match 'const shared_ptr<T>' against 'const repair_scheduler_config'
  733 | operator==(const shared_ptr<T>& x, const shared_ptr<U>& y) {
      | ^
/home/asias/src/cloudius-systems/scylla/seastar/include/seastar/core/shared_ptr.hh:740:1: note: candidate template ignored: could not match 'const shared_ptr<T>' against 'const repair_scheduler_config'
  740 | operator==(const shared_ptr<T>& x, std::nullptr_t) {
      | ^
/home/asias/src/cloudius-systems/scylla/seastar/include/seastar/core/shared_ptr.hh:747:1: note: candidate template ignored: could not match 'const shared_ptr<T>' against 'const repair_scheduler_config'
  747 | operator==(std::nullptr_t, const shared_ptr<T>& y) {
      | ^
/home/asias/src/cloudius-systems/scylla/seastar/include/seastar/core/shared_ptr.hh:754:1: note: candidate template ignored: could not match 'const lw_shared_ptr<T>' against 'const repair_scheduler_config'
  754 | operator==(const lw_shared_ptr<T>& x, std::nullptr_t) {
      | ^
/home/asias/src/cloudius-systems/scylla/seastar/include/seastar/core/shared_ptr.hh:761:1: note: candidate template ignored: could not match 'const lw_shared_ptr<T>' against 'const repair_scheduler_config'
  761 | operator==(std::nullptr_t, const lw_shared_ptr<T>& y) {
      | ^
./types/types.hh:109:1: note: candidate template ignored: could not match 'const emptyable<T>' against 'const repair_scheduler_config'
  109 | operator==(const emptyable<T>& me1, const emptyable<T>& me2) {
      | ^
./reader_permit.hh:321:6: note: candidate template ignored: could not match 'const tracking_allocator<T>' against 'const repair_scheduler_config'
  321 | bool operator==(const tracking_allocator<T>& a, const tracking_allocator<T>& b) {
      |      ^
./locator/tablets.hh:481:48: note: replace 'default' with 'delete'
  481 |     bool operator==(const tablet_map&) const = default;
      |                                                ^~~~~~~
```